src/interrupt.ts

```diff
@@ -0,0 +1,119 @@
+import type { HumanInterruptConfig } from "./types.js";
+import type { DeepAgentStateType } from "./types.js";
+import { interrupt } from "@langchain/langgraph";
+import { AIMessage } from "@langchain/core/messages";
+
+export type ToolInterruptConfig = Record<string, HumanInterruptConfig>;
+
+export interface ActionRequest {
+  action: string;
+  args: any;
+}
+
+export interface HumanInterrupt {
+  action_request: ActionRequest;
+  config: HumanInterruptConfig;
+  description: string;
+}
+
+export interface HumanResponse {
+  type: "accept" | "edit" | "ignore";
+  args?: ActionRequest;
+}
+
+export function createInterruptHook(
+  toolConfigs: ToolInterruptConfig,
+  messagePrefix: string = "Tool execution requires approval",
+): (state: DeepAgentStateType) => Promise<DeepAgentStateType> {
+  /**
+   * Create a post model hook that handles interrupts using native LangGraph schemas.
+   *
+   * Args:
+   *   toolConfigs: Record mapping tool names to HumanInterruptConfig objects
+   *   messagePrefix: Optional message prefix for interrupt descriptions
+   */
+
+  return async function interruptHook(
+    state: DeepAgentStateType,
+  ): Promise<DeepAgentStateType> {
+    const messages = state.messages || [];
+    if (!messages.length) {
+      return state;
+    }
+
+    const lastMessage = messages[messages.length - 1] as AIMessage;
+    if (!lastMessage.tool_calls || !lastMessage.tool_calls.length) {
+      return state;
+    }
```

Comment:
don't blindly assume it's an ai message
```suggestion
    const lastMessage = messages[messages.length - 1];
    if (!isAIMessage(lastMessage) || !lastMessage.tool_calls || !lastMessage.tool_calls.length) {
      return state;
    }
```